scenario.py

```diff
@@ -189,10 +198,204 @@ def __init__(self, params, experiment_path):
+        total_nb_samples = sum(final_nb_samples)
+        final_relative_nb_samples = [round(x / total_nb_samples, 2) for x in final_nb_samples]
+        # print(dict_specific)  # DEBUG
+        # print(dict_shared)  # DEBUG
+        print(final_nb_samples)  # DEBUG
+        print(final_relative_nb_samples)  # DEBUG
+
+        # Partners receive their subsets and Partner objects are instanciated
+
+        for partner in specific:
+            list_arrays_x, list_arrays_y = [], []
+            for cl in dict_specific[partner[0]]["clusters_list"]:
+                list_arrays_x.append(x_train_for_cluster[cl])
+                list_arrays_y.append(y_train_for_cluster[cl])
+
+            concat_arrays_x = np.concatenate(list_arrays_x)
+            concat_arrays_y = np.concatenate(list_arrays_y)
+            partner_x_train = concat_arrays_x[:dict_specific[partner[0]]["final_nb_samples"]]
+            partner_y_train = concat_arrays_y[:dict_specific[partner[0]]["final_nb_samples"]]
+
+            # print(partner_y_train)  # DEBUG
+
+            partner = Partner(partner_x_train, x_test, partner_y_train, y_test, str(partner[0]))
+            self.partners_list.append(partner)
+
+        for partner in shared:
+            list_arrays_x, list_arrays_y = [], []
+            for cl in dict_shared[partner[0]]["clusters_list"]:
+                list_arrays_x.append(x_train_for_cluster[cl][:dict_shared[partner[0]]["final_nb_samples_p_cluster"]])
+                list_arrays_y.append(y_train_for_cluster[cl][:dict_shared[partner[0]]["final_nb_samples_p_cluster"]])
+            partner_x_train = np.concatenate(list_arrays_x)
+            partner_y_train = np.concatenate(list_arrays_y)
+
+            # print(partner_y_train)  # DEBUG
+
+            partner = Partner(partner_x_train, x_test, partner_y_train, y_test, str(partner[0]))
+            self.partners_list.append(partner)
+
+        # Check coherence of partners_list versus partners_count
+        assert len(self.partners_list) == self.partners_count
+
+        # Check coherence of number of mini-batches versus smaller partner
+        assert self.minibatch_count <= min([len(p.x_train) for p in self.partners_list])
+
+        # Print for controlling
+        print("\n### Splitting data among partners:")
+        if self.is_advanced_split:
+            print("Advanced split performed.")
+            print("- Partners' relative nb of samples: " + str(final_relative_nb_samples))
+            print("  (versus initially configured: " + str(amounts_per_partner))
+        else:
+            print("Simple split performed.")
+        for partner in self.partners_list:
+            print("- Partner #" + str(partner.partner_id) + ":")
+            print("  - Number of samples: " + str(len(partner.x_train)) + " train samples")
+            print("  - y_train first 10 values: " + str(partner.y_train[:10]))
```

Comment:
if advanced_split:
we could show y_train unique values instead of first and last 10 values?

Reply:
Done